### DISCUSSION TITLE
Would it be a good idea to add Derive() function to System.Guid

### DISCUSSION THREAD

**Discussion by Xyncgas** [dotnet/runtime#81431] [Ideas]
As Guid is random it's commonly used to tag data, these Guids are identifier given to the records that wouldn't change, so to associate records / documents with each other, it seems natural to have this implemented in the system library, for a new Guid to deterministically be produced based on the given Guid, instead of having to do it myself by xor-ing with some other values that's also unique to these records like id (integer)

**Comment by CyrusNajmabadi** [comment]
@Xyncgas what are you asking from roslyn here?

**Reply by Xyncgas** [reply]
as far as I know, requests regarding the standard library are related to the roslyn repository

**Reply by Xyncgas** [reply]
therefore I hope it would be a good idea for you guys to put this function into the standard library so we can be calling ```System.Guid.Derive(obj)```
so what do you think

**Reply by CyrusNajmabadi** [reply]
Moving to runtime team. Roslyn is the c# compiler. Not the standard lib
